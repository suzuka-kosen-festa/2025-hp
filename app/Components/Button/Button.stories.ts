import type { Meta, StoryObj } from "@storybook/react";

import Button from "./Button";

type T = typeof Button;

const meta: Meta = {
  title: "Button",
  component: Button,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<T>;

type Story = StoryObj<T>;

export const Default: Story = {};

export default meta;
