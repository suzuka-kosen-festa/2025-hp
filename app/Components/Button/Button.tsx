import { Button as MUIButton } from "@mui/material";
import type { ReactNode } from "react";

type ButtonComponentProps = {
};

const Button = ({}: ButtonComponentProps): ReactNode => {
  return (
    <MUIButton>Hoge Fuga Piyo</MUIButton>
  );
};
export default Button;
