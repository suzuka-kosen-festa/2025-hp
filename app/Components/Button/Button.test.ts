// @vitest-environment jsdom

import { composeStories } from "@storybook/react";

import { expect, it } from "vitest";

import * as stories from "./Button.stories";

const { Default } = composeStories(stories);
it("button snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
