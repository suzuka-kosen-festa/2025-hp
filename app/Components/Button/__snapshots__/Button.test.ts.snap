// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    hoge fuga piyo ダミー
  </button>
</div>
`;
